aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -115,4 +118,25 @@ public String scheme() {
   public String toString() {
     return location;
   }
+
+  public S3URI toDirectoryPath() {
+    if (key.endsWith(PATH_DELIM)) {
+      return new S3URI(location);
+    }
+    return new S3URI(location + PATH_DELIM);
+  }
+
+  public boolean isS3DirectoryBucket() {
```

Comment:
I guess we should use the term `useS3DirectoryBucket()` which seems to fit better, since `S3URI` itself is not a directory bucket, but the URI can use a directory bucket.